### PR TITLE
Update dojo to version 6

### PR DESCRIPTION
**Pull request by agubler**
Updates Dojo to use the latest release (v6). Fixes a couple of tests to correctly assert the intended behaviour.

Supersedes: #325

### CODE REVIEW

File: libraries/dojo/package.json [hunk 1]
```diff
@@ -4,10 +4,11 @@
     "build": "npm run test; DIST=../../docs/libraries/dojo && mkdir -p $DIST && cp -r ./meta $DIST/meta && cp -r ./results $DIST/results"
```
Comment (robdodson) [inline libraries/dojo/package.json]: Would you mind updating line 3 to change the text from `@dojo/core` to `@dojo/framework`? This is used to display the version number next to the library name on the site.

**Comment by agubler** [comment]
@robdodson It would be awesome to get this merged.

**Comment by agubler** [comment]
@robdodson small ping 😄

**Comment by robdodson** [comment]
Sorry for the delay! Thank you for updating it!

**Comment by agubler** [comment]
Thank so much @robdodson

**Pull request merged**